### PR TITLE
The bookstore with the following features

### PR DESCRIPTION
**Pull request by misikir21**
1. Created a directory for your reusable components: /components.
2. Use git flow
3. Install React Router
4. Eslinters and Stylelint for standard code 
5. Address other project requirements

### CODE REVIEW

**Review by franclobo — COMMENTED** [review]
Hi @misikir21,

# Invalid Code Review Request

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that doesn't have a professional README file, please 🙏🏽 use the Microverse template.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: src/components/route/booklist.jsx [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+  const { book, deleteBook } = prop;
+  return (
+    <li key={book.id} className="book-list">
+      <h2>{book.title}</h2>
+      <p>
+        Author:
+        {book.author}
+      </p>
+      <button type="button" onClick={() => deleteBook(book.id)}>Remove</button>
+    </li>
```
Comment (Johnadibe) [inline src/components/route/booklist.jsx:15]: > - [ ] Good job @misikir21,👍 I'd like to kindly request that you reconsider adding the `<Book>` component to your project, and the BookList rendering a list of books using the `<Book>` component. Having a dedicated 'Book' component allows for better separation of concerns. Each component can focus on its specific task, making the codebase more organized and easier to maintain.😊
> 
> You can use this below as a sample:
> 
> // Book component
> ```
> const Book = ({ title, author }) => {
>   return (
>     <li>
>       <h3>Title: {title}</h3>
>       <p>Author: {author}</p>
>       <button>Remove</button> {/* Placeholder for the remove button */}
>     </li>
>   );
> };
> 
> ```
> // BookList component
> ```
> const BookList = () => {
>   // Sample book data (you can replace this with your own data)
>   const booksData = [
>     { id: 1, title: 'Book 1', author: 'Author 1' },
>     { id: 2, title: 'Book 2', author: 'Author 2' },
>     { id: 3, title: 'Book 3', author: 'Author 3' },
>   ];
> 
>   return (
>     <div>
>       <h2>Book List</h2>
>       {booksData.map((book) => (
>         <Book key={book.id} title={book.title} author={book.author} />
>       ))}
>     </div>
>   );
> };
> ```
> 
> // Books component
> ```
> const Books = () => {
>   return (
>     <div>
>       <h1>My Books</h1>
>       <BookList />
>     </div>
>   );
> };
> 
> ```

**Review by Johnadibe — CHANGES_REQUESTED** [review]
# STATUS: REQUIRED CHANGES ♻️
### Hi @misikir21👋🏽, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

mario_ladders

## Highlights
✅ Setup a new React application using Create React App✔
✅ Created a directory for reusable components✔
✅ Added 2 routes in the application in which each renders a page component✔
✅ Created a component that contains a BookList✔
✅ Created a component that contains a form to create new books✔
✅ Created a component that contains navigation to link to the created 2 routes✔

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/route/booklist.jsx (hunk 1 above)
Comment (IngDavidTM) [inline src/components/route/booklist.jsx:15]: - [ ] While you did a fantastic job @misikir21 ,👍 as mentioned by the previous reviewer I'd like to kindly request that you reconsider adding the <Book> component to your project, and the BookList rendering a list of books using the <Book> component. Having a dedicated 'Book' component allows for better separation of concerns. Each component can focus on its specific task, making the codebase more organized and easier to maintain.😊

**Review by IngDavidTM — CHANGES_REQUESTED** [review]
## Hi @misikir21 🙋‍♂️, 

Good job so far!
You have done well reaching this point of the milestone 🤝 . But there are some issues that you still need to work on to go to the next project but you are almost there!



## Top highlight 🎯
- The Pull Request follows the GitHubFlow ✅ 
- The PR title and description are professional and descriptive ✅ 
- Linters are working properly ✅ 

## Suggested changes

- [x] _Check the comments under the review._ ⬇

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/route/booklist.jsx (hunk 1 above)
Comment (misikir21) [reply src/components/route/booklist.jsx:15]: @IngDavidTM  i made the requested change but for some reason, I am unable to push it this is the last chance I get to repeat this module if I am not approved today and the time is ticking to submit.The next restart date is 2024 I can't afford to wait until then.i

**Comment by misikir21** [comment]
guys i am unable to push my changes and time is ticking I am about to die due to sterss

**Comment by Deepakdanger** [comment]
Since you have made the changes can you provide the file images or any video recording of that. @misikir21

**Review by Deepakdanger — APPROVED** [review]
Hi @misikir21 , 

### Status: Approved ✔️

Your project is complete!  There is nothing else to say other than... it's time to merge it  🚢 
Congratulations! 🎉 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- [ ] since project is working fine, you can work on the adding `BOOK` component in booklist. i can see you have issue regarding pushing & electricity also, Try creating a new branch or change the head(3-4 commits back) in your local & then pull it. It might solve the pushing issue. 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Deepakdanger** [comment]
KIndly add BOOK component before merging it.

**Pull request merged**